### PR TITLE
[Zero-Dim] Support paddle.sum/mean/loss api output 0D, test=allcase

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Description

Pcard-66984

支持以下API输出0D：
```
1. paddle.sum
2. paddle.mean
3. paddle.std
4. paddle.var
5. paddle.nanmean
6. paddle.nansum
7. paddle.nn.functional.softmax_with_cross_entropy
8. paddle.nn.functional.nll_loss
9. paddle.nn.functional.ctc_loss
10. paddle.nn.functional.kl_div
11. paddle.nn.functional.mse_loss
12. paddle.nn.functional.npair_loss
13. paddle.nn.functional.smooth_l1_loss
```

### CODE REVIEW

**Review by jeff41404 — APPROVED** [review]
LGTM

File: python/paddle/distributed/auto_parallel/engine.py [hunk 1]
```diff
@@ -511,7 +511,7 @@ def _prepare_logger(
             loss_indices = fetch_indices[group_idx]
             assert len(loss_indices) <= 1
             for idx in loss_indices:
-                logs["loss"] = outs[idx][0]
+                logs["loss"] = outs[idx]
```
Comment (aoyulong) [inline python/paddle/distributed/auto_parallel/engine.py:514]: What's the reason to remove the `[0]`?

**Review by aoyulong — APPROVED** [review]
LGTM

File: python/paddle/distributed/auto_parallel/engine.py (hunk 1 above)
Comment (zhwesky2010) [reply python/paddle/distributed/auto_parallel/engine.py:514]: outs是executor的fetch出来的loss结果，原来的fetch出来的是1D，所以这里取了下标0来切片；新的loss为0D：[]，所以就不能取下标切片了，直接当成float来用就行。

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
